code_challenges/code_wars/maximum_subarray_sum.py

```diff
@@ -0,0 +1,37 @@
+def max_sequence(sequence):
+    total = 0
+    for number in sequence:
+        if number > 0:
+            total += 1
+    if total == 0:
+        return total
+    else:
+        for i, number in enumerate(sequence):
+            j = len(sequence)
+            while j > i:
+                if sequence[j-1] > 0:
```

Comment:
What about if `sequence[i]` is negative?  If either end is negative this slice can be skipped.